### PR TITLE
[Sonic Origins] Various S3&K Codes

### PR DESCRIPTION
**Pull request by MegAmi24**
This PR adds the following Origins codes:
* Disable Ring Loss From Super/Hyper Forms
* Infinite Tails Flight
* Fix Star Post Fadeout
* Disable Bonus Stages
* Disable Cool Bonus
* Allow Amy in Classic Mode
* Allow Amy in Original Blue Spheres

### CODE REVIEW

**Comment by ActualMandM** [comment]
Please don't take two months to merge this. @thesupersonic16

File: Source/Sonic Origins/Cheats/Sonic 3 & Knuckles/Disable Ring Loss From Super Hyper Forms.hmm [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Patch "Disable Ring Loss From Super/Hyper Forms" in "Cheats/Sonic 3 & Knuckles" by "MegAmi" does "Disables losing a ring every second while in a Super or Hyper form."
+//
+    #lib "Memory"
+//
+{
+    // 2.0.2: 0x1401DF8B7
+    Memory.WriteForceJump(ScanSignature("\x0F\x8F\x9E\x01\x00\x00\xFF\x8B\xD0\x00\x00\x00", "xxxxxxxxxxxx"));
```
Comment (thesupersonic16) [inline Source/Sonic Origins/Cheats/Sonic 3 & Knuckles/Disable Ring Loss From Super Hyper Forms.hmm:7]: This does not reset the ring loss timer nor does it deactivate the form when the ring count somehow reaches 0. 6 NOP at 0x1401DF8BD might be more desireable?

File: Source/Sonic Origins/Gameplay/Sonic 3 & Knuckles/Disable Cool Bonus.hmm [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Patch "Disable Cool Bonus" in "Gameplay/Sonic 3 & Knuckles" by "MegAmi" does "Disables the leftover Cool score bonus from Sonic Mania."
+//
+    #lib "Memory"
+//
+{
+    // 2.0.2: 0x1401FF215
+    Memory.WriteForceJump(ScanSignature("\x75\x55\xC7\x81\x00\x00\x00\x00\x10\x27\x00\x00", "xxxx????xxxx"));
```
Comment (thesupersonic16) [inline Source/Sonic Origins/Gameplay/Sonic 3 & Knuckles/Disable Cool Bonus.hmm:7]: Setting the globals might also be an option, but this is fine.

**Review by thesupersonic16 — APPROVED** [review]
LGTM

I left some comments incase you want to make any changes, but what you wrote is fine for me. Let me know when you are ready for the merge.

I trust that the star post fix code works lol

File: Source/Sonic Origins/Cheats/Sonic 3 & Knuckles/Disable Ring Loss From Super Hyper Forms.hmm (hunk 1 above)
Comment (MegAmi24) [reply Source/Sonic Origins/Cheats/Sonic 3 & Knuckles/Disable Ring Loss From Super Hyper Forms.hmm]: I figured this wouldn't be too much of an issue, but I've changed it if that's what you think would be better.

File: Source/Sonic Origins/Gameplay/Sonic 3 & Knuckles/Disable Cool Bonus.hmm (hunk 2 above)
Comment (MegAmi24) [reply Source/Sonic Origins/Gameplay/Sonic 3 & Knuckles/Disable Cool Bonus.hmm:7]: This is easier for me than trying to dig for the variable's offset in the globals, plus a patch like this is better than constantly writing the value imo.

**Review by thesupersonic16 — APPROVED** [review]
LGTM

**Pull request merged**